Add Beats known issue caused by credentials error with S3+SQS (#33013)
elastic/beats: CHANGELOG.asciidoc

=== Beats version 8.4.1
https://github.com/elastic/beats/compare/v8.4.0\...v8.4.1[View commits]

==== Known Issue 

// tag::credentials-error[]

*Filebeat*

Filebeat agents configured to read from AWS inputs may return an error similar to the following:

[source,shell]
----
sqs ReceiveMessage failed: operation error SQS: ReceiveMessage, https response
error StatusCode: 403, RequestID: cb57783a-505f-5099-9160-23b8eea8ddbb,
api error SignatureDoesNotMatch: Credential should be scoped to a valid region. 
----

This error was introduced by a breaking change in the AWS library.

Suggested resolution: In the Filebeat configuration, if an AWS input or module
configuration sets `endpoint` to a non empty string, set it to an empty string
instead. Also make sure the default AWS region is set in an environment
variable, credentials or instance profile, or in the `default_region` setting in
the configuration. For example:

[source,yaml]
----
filebeat.inputs:
- type: aws-s3

> ...
>   endpoint: ""
>   default_region: us-east-1
> ----
> 

Author: ```suggestion
...
  endpoint: "" <1>
  default_region: us-east-1
----
<1> You can set this value to an empty string or remove the configuration setting.
```

Or for modules:

[source,yaml]
----
s3access:
    enabled: false
...

>     var.endpoint: ""
>     var.default_region: us-east-1
> ----
> 

Author: ```suggestion
    var.endpoint: "" <1>
    var.default_region: us-east-1
----
<1> You can set this value to an empty string or remove the configuration setting

```

=== Beats version 8.4.1
https://github.com/elastic/beats/compare/v8.4.0\...v8.4.1[View commits]

==== Known Issue 

// tag::credentials-error[]

*Filebeat*

> Filebeat agents configured to read from S3 or SQS may return the following

Reviewer: ```suggestion
Filebeat agents configured to read from S3 input or S3 input with SQS notification setup may return the following
```

Author: I applied Andrea's recommendation to make this more generic instead.

> Filebeat agents configured to read from S3 or SQS may return the following

Reviewer 2: leaving here similar comment as ones at https://github.com/elastic/observability-docs/pull/2168
I would be "less" specific: meaning is a problem related to any AWS metricset/fileset and not specific to S3 and SQS.

> Filebeat agents configured to read from S3 or SQS may return the following

Author: ```suggestion
Filebeat agents configured to read from AWS inputs may return the following
```

> Filebeat agents configured to read from S3 or SQS may return the following
> error:

Author: ```suggestion
Filebeat agents configured to read from AWS inputs may return an error similar to the following:
```

[source,shell]
----

> sqs ReceiveMessage failed: operation error SQS: ReceiveMessage, https response

Reviewer 2: beware that the error message might differ for some users: it's anyway most of the time related to authentication and/or region

error StatusCode: 403, RequestID: cb57783a-505f-5099-9160-23b8eea8ddbb,
api error SignatureDoesNotMatch: Credential should be scoped to a valid region. 
----

This error was introduced by a breaking change in the AWS library.

Suggested resolution: In the Filebeat configuration, if an AWS input or module

> configuration sets `endpoint` to `amazonaws.com`, set it to an empty string

Reviewer: ```suggestion
configuration sets `endpoint` to `amazonaws.com`, set it to an empty string or remove this config parameter
```

> configuration sets `endpoint` to `amazonaws.com`, set it to an empty string

Reviewer 2: ```suggestion
configuration sets `endpoint` to a non empty string, set it to an empty string
```

instead. Also make sure the default AWS region is set in an environment

> variable, credentials or instance profile, or in the `default_region` setting in

Reviewer 2: 👍

the configuration. For example:

[source,yaml]
----

> filebeat.inputs:

Reviewer: or can also be:
```
filebeat.inputs:
- type: aws-s3
...
  default_region: us-east-1
```

...
  endpoint: "" <1>
  default_region: us-east-1
----

> <1> You can set this value to an empty string or remove the configuration setting.
> Or for modules:

Author: ```suggestion
<1> You can set this value to an empty string or remove the configuration setting.

Or for modules:
```
